packages/components/src/navigator/types.ts

```diff
@@ -57,8 +58,19 @@ export type NavigatorScreenProps = {
 	children: ReactNode;
 };
 
+export type NavigatorBackButtonHookProps = ButtonAsButtonProps & {
+	/**
+	 * Whether we should navigate to the parent screen.
+	 *
+	 * @default 'false'
+	 */
+	goToParent?: boolean;
+};
+
 export type NavigatorBackButtonProps = ButtonAsButtonProps;
 
+export type NavigatorToParentButtonProps = ButtonAsButtonProps;
```

Comment:
Nitty: reshuffling types for better inheritance

```suggestion
export type NavigatorBackButtonProps = ButtonAsButtonProps;

export type NavigatorBackButtonHookProps = NavigatorBackButtonProps & {
	/**
	 * Whether we should navigate to the parent screen.
	 *
	 * @default 'false'
	 */
	goToParent?: boolean;
};

export type NavigatorToParentButtonProps = NavigatorBackButtonProps;
```